api/tests/unit/application/campaign/campaign-controller_test.js

```diff
@@ -0,0 +1,178 @@
+const { sinon, expect, factory } = require('../../../test-helper');
+
+const campaignController = require('../../../../lib/application/campaigns/campaign-controller');
+const usecases = require('../../../../lib/domain/usecases');
+const campaignRepository = require('../../../../lib/infrastructure/repositories/campaign-repository');
+const userRepository = require('../../../../lib/infrastructure/repositories/user-repository');
+const campaignSerializer = require('../../../../lib/infrastructure/serializers/jsonapi/campaign-serializer');
+const { UserNotAuthorizedToCreateCampaignError, EntityValidationError } = require('../../../../lib/domain/errors');
+
+describe('Unit | Application | Controller | Campaign', () => {
+
+  describe('#save', () => {
+
+    let replyStub;
+    let codeStub;
+
+    beforeEach(() => {
+      sinon.stub(usecases, 'createCampaign');
+      sinon.stub(campaignSerializer, 'deserialize');
+      sinon.stub(campaignSerializer, 'serialize');
+      codeStub = sinon.stub();
+      replyStub = sinon.stub().returns({
+        code: codeStub,
+      });
+    });
+
+    afterEach(() => {
+      usecases.createCampaign.restore();
+      campaignSerializer.deserialize.restore();
+      campaignSerializer.serialize.restore();
+    });
+
+    it('should call the use case to create the new campaign', () => {
+      // given
+      const userId = 1245;
+      const request = { auth: { credentials: { userId } } };
+
+      const deserializedCampaign = factory.buildCampaign({ id: NaN, code: '', creatorId: '' });
+      campaignSerializer.deserialize.returns(deserializedCampaign);
+
+      const createdCampaign = factory.buildCampaign();
+      usecases.createCampaign.resolves(createdCampaign);
+
+      // when
+      const promise = campaignController.save(request, replyStub);
+
+      // then
+      return promise.then(() => {
+        const createCampaignArgs = usecases.createCampaign.firstCall.args[0];
```

Comment:
Au vu du titre du test, j'aurais ajouté un expect(usecases.createCampaign).to.be.called.once

Reply:
C'est vrai. Cette vérif est fait implicitement quand on regarde les paramètres de l'appel. Mais c'est mieux de l'expliciter.